### ISSUE TITLE
cover_path params not present in Rails Controller

### ISSUE THREAD

**Issue by jnimety** [Studiosity/grover#40]
I might be doing something wrong here, but I am unable to see my cover_path params when the request is initiated by grover. When I request the page directly via browser the params are visible.
Is this a known limitation?

`front_cover_path: pdf_cover_page_path(title: "Title", subtitle: "Subtitle"),`

```
[5, 16] in app/controllers/pdf_cover_pages_controller.rb
    5:   def show
    6:     byebug
=>  7:     @title = pdf_cover_page_params[:title]
    8:     @subtitle = pdf_cover_page_params[:subtitle]
    9:   end
   10:
   11:   private
   12:
   13:   def pdf_cover_page_params
   14:     params.permit(:title, :subtitle)
   15:   end
   16: end
(byebug) request.params
{"controller"=>"pdf_cover_pages", "action"=>"show"}
(byebug) request.url
"https://host.redacted.net/pdf_cover_page?subtitle=Subtitle&title=Title"
(byebug) params
<ActionController::Parameters {"controller"=>"pdf_cover_pages", "action"=>"show"} permitted: false>
```

**Comment by abrom** [comment]
No, looks like a bug to me! Rails is doing some caching of the request parameters in the `env` which results in the params being what ever they were for the original request..

However the cache can be busted with the following:

```
request.delete_header 'action_dispatch.request.parameters'
request.delete_header 'action_dispatch.request.query_parameters'
puts params
=> {"title"=>"Title", "subtitle"=>"Subtitle", "controller"=>"pdf_cover_pages", "action"=>"show"}
```

As a short term measure you could put those `delete_header` calls into your cover page controller in a `before_action` callback. 

Taking a look at other places that Rails messes about with the above headers I can see the following method in their `TestCase` class:
```
        def scrub_env!(env)
          env.delete_if { |k, v| k =~ /^(action_dispatch|rack)\.request/ }
          env.delete_if { |k, v| k =~ /^action_dispatch\.rescue/ }
          env.delete "action_dispatch.request.query_parameters"
          env.delete "action_dispatch.request.request_parameters"
          env["rack.input"] = StringIO.new
          env.delete "CONTENT_LENGTH"
          env.delete "RAW_POST_DATA"
          env
        end
```

The likely solution will be to modify Grover to take Rails' lead on what they expect to be scrubbed between requests. Leave it with me, I'll tag you when I've got a PR together

**Comment by abrom** [comment]
Is a bit strange that they have the 3rd and 4th lines (explicitly deleting the query/request parameters) given the first line should delete all keys starting with `action_dispatch.request` ?! 

FYI the caching is coming from the Actionpack gem in action_dispatch/http/parameters.rb in the `parameters` method and then subsequently action_dispatch/http/request.rb in the `GET` and `POST` methods. The above methods check the respective env keys, as a cache, then if needed build the cache..

**Comment by jnimety** [comment]
@abrom thanks for the quick response and short term work around!

**Comment by abrom** [comment]
See #41 for the permanent fix. I'll hold off merging/deploying until I've heard back from you RE testing the branch on your setup

**Comment by abrom** [comment]
#41 fixes this (released as v0.11.1)
